src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -1425,6 +1462,12 @@ private Expression BindCustomMethodExpressionOrNull(SingleValueFunctionCallNode
         /// <returns>The LINQ <see cref="Expression"/> created.</returns>
         private Expression BindSingleValueNode(SingleValueNode node)
         {
+            CountNode countNode = node as CountNode;
+            if (countNode != null)
+            {
+                return BindCountNode(countNode);
+            }
+
             switch (node.Kind)
```

Comment:
Any reason this is not in the switch statement for case: QueryNodeKind.Count?

Reply:
No reason. Fixed.